credentials/generate-revocation-set.py

```diff
@@ -198,27 +205,98 @@ def fetch_crl_from_url(url: str, timeout: int) -> x509.CertificateRevocationList
         logging.error('Failed to fetch a valid CRL')
 
 
-class DCLDClient:
+class DCLDClientInterface:
     '''
-    A client for interacting with DCLD using either the REST API or command line interface (CLI).
+    An interface for interacting with DCLD.
 
     '''
 
-    def __init__(self, use_rest: bool, dcld_exe: str, production: bool, rest_node_url: str):
+    def get_revocation_points(self) -> list[dict]:
+        '''
+        Get revocation points from DCL
+
+        Returns
+        -------
+        list[dict]
+            List of revocation points
+        '''
+        raise NotImplementedError
+
+    def get_revocations_points_by_skid(self, issuer_subject_key_id) -> list[dict]:
+        '''
+        Get revocation points by subject key ID
+
+        Parameters
+        ----------
+        issuer_subject_key_id: str
+            Subject key ID
+
+        Returns
+        -------
+        list[dict]
+            List of revocation points
+        '''
+        raise NotImplementedError
+
+    def get_certificate(self, issuer_name_b64: str, akid_hex: str) -> tuple[bool, Optional[x509.Certificate]]:
+        '''
+        Get certificate from DCL
+        '''
+        raise NotImplementedError
+
+    def get_first_approved_certificate(self, response: dict, akid_hex: str) -> tuple[bool, Optional[x509.Certificate]]:
+        '''
+        Get first approved certificate from DCL resposne.
+        '''
+        if response["approvedCertificates"]["certs"][0]["pemCert"]:
+            issuer_certificate = x509.load_pem_x509_certificate(
+                bytes(response["approvedCertificates"]["certs"][0]["pemCert"], 'utf-8'))
+        else:
+            raise requests.exception.NotFound(f"No certificate found for {akid_hex}")
+        return response["approvedCertificates"]["certs"][0]["isRoot"], issuer_certificate
```

Comment:
It may be cleaner to capture the first element and then return that. That way, we do not need to check for doing the same thing on both this line and the one above in the if statement. WDYT?

Reply:
Ive decided to continue with the tuple returned as it helps simplify things when the local file DCLclient is added in the next CL.

Reply:
LGTM